nexus/completion.go

```diff
@@ -33,10 +34,13 @@ type OperationCompletion interface {
 // OperationCompletionSuccessful is input for [NewCompletionHTTPRequest], used to deliver successful operation results.
 type OperationCompletionSuccessful struct {
 	// Header to send in the completion request.
-	Header http.Header
-	// Body to send in the completion HTTP request.
-	// If it implements `io.Closer` it will automatically be closed by the client.
-	Body io.Reader
+	// Note that this is a Nexus header, not an HTTP header.
+	Header Header
+
+	// A [Reader] that may be directly set on the completion or constructed when instantiating via
+	// [NewOperationCompletionSuccessful].
+	// Automatically closed when the completion is delivered.
+	Reader *Reader
```

Comment:
Doesn't `Reader` already have a header on it?

Reply:
Yes, that's a content header though, not a Header for the entire request.